### ISSUE TITLE
Error without any stack trace

### ISSUE THREAD

**Issue by Mutefish0** [denoland/deno_core#313]
I often encounter the following error in my program, but I don't know where the error is coming from.

<img width="901" alt="截屏2023-11-08 下午6 52 22" src="https://github.com/denoland/deno_core/assets/15227926/ade21821-f58a-4c1d-b8bb-5e98ae440e7f">

based on this error message, there seems to be some bug in 01_core.js:

<img width="429" alt="截屏2023-11-08 下午7 05 03" src="https://github.com/denoland/deno_core/assets/15227926/70f7c5ae-53f5-4f47-bfe7-4af86146164b">

`getPromise(promiseId)` could potentially return undefined, but this situation is not being handled here.

deno version: 
```
deno 1.37.2 (release, x86_64-unknown-linux-gnu)
v8 11.8.172.13
typescript 5.2.2
```

**Comment by bartlomieju** [comment]
Do you have a reproduction code that you could share? This situation should never happen.

**Comment by Mutefish0** [comment]
I can't reproduce it since I don't known what cause the error. It's a complex app running on my server, this error occurs approximately once a day, sometimes not at all.

But I can provide a brief overview of the app's architecture.

<img width="678" alt="截屏2023-11-08 21 19 18" src="https://github.com/denoland/deno_core/assets/15227926/b6ce5d7d-5d4a-4f5a-8b08-e9b061079e80">

First, the main program launches a worker, 'worker a', which initiates a WebSocket connection to subscribe to data from another website. It also listens on port 12002 using `Deno.listen`. Then, the main program starts several other workers, like 'worker 1' and 'worker 2,' which connect to port 12002 using `Deno.connect`. They use a `while(1) await conn.read(buffer)` loop to receive data from 'worker a.' Each time 'worker a' receives data from the WebSocket, it forwards the data to 'worker 1' and 'worker 2' through `conn.write`. Then at some point, 'worker a' will inexplicably generate the above error, with no additional error information, according to the app's log message, there was even no data transmission when this error occurred.

**Comment by mmastrac** [comment]
I'll take a look at this -- @Mutefish0 could you give me an idea of packet rates to help diagnose? ie: how many websocket messages per day, roughly?

**Comment by mmastrac** [comment]
I tried setting nextPromiseId to `0xffffffff` in 01_core.js and I get the same result. Looks like it's a promise roll-over problem.

**Comment by mmastrac** [comment]
We've wallpapered over the problem a bit but it still exists in a slightly different form. Opened https://github.com/denoland/deno_core/issues/319 for follow-up but you should hopefully not see daily crashes.

**Comment by Mutefish0** [comment]
> I'll take a look at this -- @Mutefish0 could you give me an idea of packet rates to help diagnose? ie: how many websocket messages per day, roughly?

about 1-10 messages per second, but there are up to 20 such modules in my app. I'm wondering if they all share the same 
32-bit promise IDs and could cause roll-over problem despite they are running at different workers ?

**Comment by mmastrac** [comment]
I discovered another possible problem that might be causing this and it'll be fixed in #323.
